### PR TITLE
VIH-7283 assign ejudiciary judge acccount

### PR DESCRIPTION
**Pull request by valentinabennett**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-7283

### Change description ###

assign judge account and validate 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ X] Yes
[ ] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts [hunk 1]
```diff
@@ -349,6 +376,19 @@ export class AssignJudgeComponent extends BookingBaseComponent implements OnInit
         this.expanded = !this.expanded;
     }
 
+    validateJudgeAndJohMembers(): boolean {
+        if (this.hearing?.participants.length > 0 && this.judgeName.value) {
+            const johMembers = this.hearing.participants.filter(
+                x => x.hearing_role_name === 'Panel Member' || x.hearing_role_name === 'Winger'
+            );
+            if (johMembers?.length > 0) {
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts:384]: could use `.some()` to check if its empty or not

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts [hunk 2]
```diff
@@ -91,7 +101,7 @@ export class AssignJudgeComponent extends BookingBaseComponent implements OnInit
         this.populateFormFields(existingJudge);
 
         this.form = this.fb.group({
-            judgeName: [this.judge.display_name, Validators.required],
+            judgeName: [this.judge.email, Validators.required],
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts:104]: is calling this judgeName still correct now it contains the email?

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-validation.service.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            email &&
+            email.length > 0 &&
+            email.length < 256 &&
+            pattern.test(email.toLowerCase()) &&
+            email.toLowerCase().indexOf(invalidPattern) < 0;
+        return isValidEmail;
+    }
+
+    hasCourtroomAccountPattern(email: string, invalidPattern: string): boolean {
+        return email?.toLowerCase().indexOf(invalidPattern) > -1;
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-validation.service.ts:23]: does the pattern also need to be tolowered?

Comment (levibutler) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-validation.service.ts:23]: coming from `this.invalidPattern = this.route.snapshot.data['emailPattern'];`
think it should probably be tolowered also

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-validation.service.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    validateEmail(email: string, invalidPattern: string): boolean {
+        /* tslint:disable: max-line-length */
+        const pattern = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.(?:[a-zA-Z0-9](?:\.[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
+
+        const isValidEmail =
+            email &&
+            email.length > 0 &&
+            email.length < 256 &&
+            pattern.test(email.toLowerCase()) &&
+            email.toLowerCase().indexOf(invalidPattern) < 0;
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-validation.service.ts:18]: this.hasCourtroomAccountPattern(email, invalidPattern)

**Review by levibutler — COMMENTED** [review]
couple little comments

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts (hunk 2 above)
Comment (valentinabennett) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts:104]: This is the contact email, the second one

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts (hunk 1 above)
Comment (valentinabennett) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts:384]: Done

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-validation.service.ts (hunk 3 above)
Comment (valentinabennett) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-validation.service.ts:23]: taken off

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-validation.service.ts (hunk 4 above)
Comment (valentinabennett) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-validation.service.ts:18]: This is this.hasCourtroomAccountPattern(email, invalidPattern) used for recognising the judiciary account and courtroom account

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-pattern-resolver.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+@Injectable()
+export class EmailPatternResolver implements Resolve<string> {
+    constructor(private configService: ConfigService) {}
+
+    resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<string> {
+        return from(this.getData());
+    }
+
+    async getData() {
+        const settings = await this.configService.getClientSettings().toPromise();
```
Comment (levibutler) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-pattern-resolver.ts:15]: will this work?

Comment (levibutler) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-pattern-resolver.ts:15]: the promise may not be resolved?

Comment (levibutler) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-pattern-resolver.ts:15]: @phillipjchambers

**Review by levibutler — COMMENTED** [review]
lgtm, asked phill to confirm something

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-pattern-resolver.ts (hunk 5 above)
Comment (valentinabennett) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-pattern-resolver.ts:15]: It does, I tested it

Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-pattern-resolver.ts:15]: The route will not be changed _until_ this route is resolved

**Review by phillipjchambers — APPROVED** [review]

**Pull request merged**